src/lager_transform.erl

```diff
@@ -75,138 +77,148 @@ walk_clauses(Acc, [{clause, Line, Arguments, Guards, Body}|T]) ->
+                                           %% The trace attributes will be a list of tuples, so check
+                                           %% for that.
+                                           case {element(1, Arg1), Arg1} of
+                                               {_, {cons, _, {tuple, _, _}, _}} ->
+                                                   {concat_lists(Arg1, DefaultAttrs),
+                                                    Arg2, {atom, Line, none}};
+                                               {Type, _} when Type == var;
+                                                              Type == lc;
+                                                              Type == call;
+                                                              Type == record_field ->
+                                                   %% crap, its not a literal. look at the second
+                                                   %% argument to see if it is a string
+                                                   case Arg2 of
+                                                       {string, _, _} ->
+                                                           {concat_lists(Arg1, DefaultAttrs),
+                                                            Arg2, {atom, Line, none}};
+                                                       _ ->
+                                                           %% not a string, going to have to guess
+                                                           %% it's the argument list
+                                                           {DefaultAttrs, Arg1, Arg2}
+                                                   end;
+                                               _ ->
+                                                   {DefaultAttrs, Arg1, Arg2}
+                                           end;
+                                       [Attrs, Format, Args] ->
+                                           {concat_lists(Attrs, DefaultAttrs), Format, Args}
+                                   end,
+    %% Generate some unique variable names so we don't accidentaly export from case clauses.
+    %% Note that these are not actual atoms, but the AST treats variable names as atoms.
+    LevelVar = make_varname("__Level", Line),
+    TracesVar = make_varname("__Traces", Line),
+    PidVar = make_varname("__Pid", Line),
+    %% Wrap the call to lager:dispatch_log/6 in case that will avoid doing any work if this message is not elegible for logging
+    %% See lager.erl (lines 89-100) for lager:dispatch_log/6
+    %% case {whereis(Sink), whereis(?DEFAULT_SINK), lager_config:get({Sink, loglevel}, {?LOG_NONE, []})} of
+    {'case',Line,
+         {tuple,Line,
+                [{call,Line,{atom,Line,whereis},[{atom,Line,SinkName}]},
+                 {call,Line,{atom,Line,whereis},[{atom,Line,lager_event}]}, %% <--lager_event is ?DEFAULT_SINK
```

Comment:
I wasn't sure of the order of transform -> preprocessor or preprocessor -> transform.  It seemed safer to use the naked atom here. But we should give the macro a shot, I suppose.